amqp/src/main/java/org/axonframework/amqp/eventhandling/DefaultAMQPMessageConverter.java

```diff
@@ -112,21 +114,97 @@ public AMQPMessage createAMQPMessage(EventMessage<?> eventMessage) {
         } else {
             return Optional.of(new GenericEventMessage<>(message, () -> DateTimeUtils.parseInstant(timestamp)));
         }
     }
 
+    /**
+     * Builder class to instantiate a {@link DefaultAMQPMessageConverter}.
+     * The {@link RoutingKeyResolver} is defaulted to a {@link PackageRoutingKeyResolver} and the {@code durable} field
+     * defaults to {@code true}. The {@link Serializer} is a <b>hard requirement</b> and thus should be provided.
+     */
+    public static class Builder {
+
+        private Serializer serializer;
+        private RoutingKeyResolver routingKeyResolver = new PackageRoutingKeyResolver();
+        private boolean durable = true;
+
+        /**
+         * Sets the serializer to serialize the Event Message's payload and Meta Data with.
+         *
+         * @param serializer The serializer to serialize the Event Message's payload and Meta Data with
+         * @return the current Builder instance, for a fluent interfacing
+         */
+        public Builder serializer(Serializer serializer) {
+            assertThat(serializer,
+                       Objects::nonNull,
+                       () -> new AxonConfigurationException("Serializer may not be null"));
+            this.serializer = serializer;
+            return this;
+        }
+
+        /**
+         * Sets the strategy to use to resolve routing keys for Event Messages. Defaults to a
+         * {@link PackageRoutingKeyResolver}.
+         *
+         * @param routingKeyResolver The strategy to use to resolve routing keys for Event Messages
+         * @return the current Builder instance, for a fluent interfacing
+         */
+        public Builder routingKeyResolver(RoutingKeyResolver routingKeyResolver) {
+            assertThat(routingKeyResolver,
+                       Objects::nonNull,
+                       () -> new AxonConfigurationException("RoutingKeyResolver may not be null"));
+            this.routingKeyResolver = routingKeyResolver;
+            return this;
+        }
+
+        /**
+         * Sets a {@code boolean} specifying whether to request durable message dispatching. Defaults to {@code true},
+         * thus toggling durable message dispatching on.
+         *
+         * @param durable a {@code boolean} specifying whether to request durable message dispatching
+         * @return the current Builder instance, for a fluent interfacing
+         */
+        public Builder durable(boolean durable) {
+            this.durable = durable;
+            return this;
+        }
+
+        /**
+         * Initializes a {@link DefaultAMQPMessageConverter} as specified through this Builder.
+         *
+         * @return a {@link DefaultAMQPMessageConverter} as specified through this Builder
+         */
+        public DefaultAMQPMessageConverter build() {
+            return new DefaultAMQPMessageConverter(this);
+        }
+
+        private void validate() {
+            assertThat(serializer,
```

Comment:
Why not extract `assertThat...` into separate methods: `assertRoutingKeyResolver` for example. If the requirement is not fulfilled, a message could be provided to be used for an `AxonConfigurationException` (I would argue that we could use the same message for both purposes: invoking the `build` method and `validate` method).

Reply:
Added a `BuilderUtils`, containing an `assertThat(T, Predicate<T>, String)` and an `assertNonNull(T, String)`, and used it in all the assert spots.
This should simplify the whole look.